src/BulkTools/HTTPBulkToolsResponse.php

```diff
@@ -369,7 +369,7 @@ protected function outputBody()
     public function shutdown()
     {
         $error = error_get_last();
-        if ($error !== null ) {
+        if ($error !== null && $error['type'] !== E_USER_DEPRECATED) {
```

Comment:
There's a lot of other warnings and notices being thrown aside from the deprecation warning. Even so, in a scenario where multiple errors are thrown, you'll miss all of them if the last one is a deprecation warning.

You're also exposing an internal error message even in a live environment, which is a security risk.

It's not even clear to me the the error message ends up being displayed in a useful way to the user in the front end.

Basically, this approach is incredibly shaky and I think we would be better off gutting the shut down method instead.

Reply:
https://github.com/colymba/GridFieldBulkEditingTools/pull/205 😄